internal/util/errors.go

```diff
@@ -0,0 +1,12 @@
+//
+// Copyright Strimzi authors.
+// License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+//
+
+// Package util contains some utility functions
+package util
+
+const (
+	ErrEof                   = "EOF"
```

Comment:
I don't know sarama, but does it expose the underlying network err codes, so you can do something more idiomatic.  Relying of string comparisons feels weak.

```
errors.Is(err, syscall.ECONNRESET)
```

Reply:
I know, the problem is that Sarama seems to mask that error but it's worth trying, maybe I did something bad.

Reply:
I made a mistake when checking the error so `errors.Is` was not working.
Right now I tried with `errors.Is(err, io.EOF)` because I am able to reproduce it and it works.
See my new changes.